nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/ui/web/auth/NuxeoAuthenticationFilter.java

```diff
@@ -299,14 +299,25 @@ protected Principal doAuthenticate(CachableUserIdentificationInfo cachableUserId
 
             logAuthenticationAttempt(cachableUserIdent.getUserInfo(), true);
         } catch (LoginException e) {
-            log.info("Login failed for " + cachableUserIdent.getUserInfo().getUserName());
+            if (log.isInfoEnabled()) {
+                log.info(String.format("Login failed for %s on request %s",
+                        cachableUserIdent.getUserInfo().getUserName(), httpRequest.getRequestURI()));
+            }
+            log.debug(e, e);
             logAuthenticationAttempt(cachableUserIdent.getUserInfo(), false);
             Throwable cause = e.getCause();
             if (cause instanceof DirectoryException) {
                 Throwable rootCause = ExceptionUtils.getRootCause(cause);
                 if (rootCause instanceof NamingException
                         && rootCause.getMessage().contains("LDAP response read timed out")
                         || rootCause instanceof SocketException) {
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format(
+                                "Exception root cause is either a NamingException with \"LDAP response read timed out\""
+                                        + " or a SocketException, setting the status code to %d,"
+                                        + " more relevant than an illegitimate 401.",
+                                HttpServletResponse.SC_GATEWAY_TIMEOUT));
+                    }
```

Comment:
we could also trace the rootCause in the debug so we know exactly if it si a ldap or socket exception

Reply:
Yes, the root cause will be there, this is just a convenience to clearly explicit the fact that the return code is specific regarding the nature of the root cause.